### ISSUE TITLE
Gradients do not exist for variables

### ISSUE THREAD

**Issue by nialloh23** [tensorflow/recommenders#104]
I'm trying to train a model that includes candidate side features (e.g. for a given fashion style I've tried to include it's shape. This is a [google collab file ](https://colab.research.google.com/drive/1qRCU0zeqf6KMtmUevZdHSjFkF4bnqtNW?usp=sharing) in which the example can be run.

This is a high level summary of how my input data is configured:  
```
interactions = interactions.map(lambda x: {
    "user_id": x["user_id"],
    "style": x["style"],
    "high_level_shape": x["high_level_shape"]
})
styles = styles.map(lambda x: {
    "style": x["style"],
    "high_level_shape": x["high_level_shape"]
})
```   
However when I run the two tower model I get the following TF warning which appears concerning:  

`WARNING:tensorflow:Gradients do not exist for variables ['embedding_1/embeddings:0', 'embedding_2/embeddings:0', 'sequential_3/dense/kernel:0', 'sequential_3/dense/bias:0'] when minimizing the loss. `  

All of the tutorial examples provided have only one candidate feature (e.g. movie_title) so I'm not sure if the approach I'm taking is exactly correct or if there is a problem with my model configuration?

The other warning message I get on most of runs is from the way I pass in my dataset to the keras fit method.   
`
WARNING:tensorflow:Layers in a Sequential model should only have a single input tensor, but we receive a <class 'dict'> input: {'style': <tf.Tensor 'IteratorGetNext:1' shape=(None,) dtype=string>, 'high_level_shape': <tf.Tensor 'IteratorGetNext:0' shape=(None,) dtype=string>} 

Consider rewriting this model with the Functional API.
/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/functional.py:543: UserWarning: Input dict contained keys ['style', 'high_level_shape'] which did not match any model input. They will be ignored by the model.
  [n for n in tensors.keys() if n not in ref_input_names])
`

From looking at the [keras documentation](https://keras.io/api/models/model_training_apis/) it appears passing in a TF dataset as a dictionary of multiple key value pairs gets picked up as a single X (input dataset) with no target -> ({"x0": x0, "x1": x1, "x3":x3}. From my understanding of how TF reccomenders work this is what we want to do as it is assumed that every interaction is positive and as such we don't explicity pass in a target value. Is this warning expected and does it have anything to do with the other gradients warning above? I noticed that the tutorial examples provided also have similar warnings that don't seem problematic so this may be totally unrelated.

Apologies for the long winded query but have tried to provide as much context as possible. Thanks for any help or guidance on the above.

**Comment by nialloh23** [comment]
For anyone who has similar problems in the future:
I ended up finding an error in my configuration where I was using the user/query model for both my query and candidate embeddings:

      user_embeddings = **self.user_model(**features["user_id"])
      positive_style_embeddings = **self.user_model**({
            "style": features["style"],
            "high_level_shape": features["high_level_shape"],
        })
